sync/class.jetpack-sync-module-posts.php

```diff
@@ -42,6 +44,84 @@ public function init_listeners( $callable ) {
 		// listen for meta changes
 		$this->init_listeners_for_meta_type( 'post', $callable );
 		$this->init_meta_whitelist_handler( 'post', array( $this, 'filter_meta' ) );
+
+		add_action( 'jetpack_sync_import_end', $callable );
+
+		// Movable type, RSS, Livejournal
+		add_action( 'import_done', array( $this, 'sync_import_done' ) );
+
+		// WordPress, Blogger, Livejournal, woo tax rate
+		add_action( 'import_end', array( $this, 'sync_import_end' ) );
+	}
+
+	public function sync_import_done( $importer ) {
+		// We already ran an send the import
+		if ( $this->import_end ) {
+			return;
+		}
+		/**
+		 * Sync Event that tells that that the import is finished
+		 *
+		 * @since 5.0.0
+		 *
+		 * $param string $importer 
+		 */
+		do_action( 'jetpack_sync_import_end', $importer );
+		$this->import_end = true;
+	}
+
+	public function sync_import_end() {
+		// We already ran an send the import
+		if ( $this->import_end ) {
+			return;
+		}
+
+		$this->import_end = true;
+		$importer = 'unknown';
+		$backtrace = wp_debug_backtrace_summary(null, 0, false );
+		if ( $this->is_blogger_importer( $backtrace ) ) {
+			$importer = 'blogger';
+		}
+
+		if ( 'unknown' === $importer && $this->is_woo_tax_rate_importer( $backtrace ) ) {
+			$importer = 'woo-tax-rate';
+		}
+
+		if ( 'unknown' === $importer && $this->is_wp_importer( $backtrace ) ) {
+			$importer = 'wordpress';
+		}
+
+		/**
+		 * Documented already: Sync Event that tells that that the import is finished
+		 */
+		do_action( 'jetpack_sync_import_end', $importer );
+	}
+	
+	private function is_blogger_importer( $backtrace ) {
```

Comment:
Consider replacing the is_blogger_importer, is_woo_tax_rate_importer, and is_wp_importer methods with a single is_importer( $test_importer, $backtrace) method that takes a search string as a parameter in addition to the backtrace so that you can reuse the same method and eliminate duplicate code.